Kernel/Substitution.hpp

```diff
@@ -12,61 +12,85 @@
  * Defines class Substitution.
  *
  * @since 08/07/2007 Manchester, flight Manchester-Cork
- * @since 30/12/2007 Manchester, reimplemented from scratch using a skip list
- * like structure
+ * @since 30/12/2007 Manchester, reimplemented from scratch using a skip list like structure
+ * @since 28/05/2025 Southampton, simplify, inline methods, conform to interfaces
  */
 
 #ifndef __Substitution__
 #define __Substitution__
 
-#include "Lib/DHMap.hpp"
-#include "Lib/Environment.hpp"
-
 #include "Lib/Allocator.hpp"
+#include "Lib/DHMap.hpp"
 
 #include "Term.hpp"
 
 namespace Kernel {
 
-using namespace Lib;
-
 /**
  * The class Substitution implementing substitutions.
  * @since 30/12/2007 Manchester
+ *
+ * This is a simple map from variables to terms.
+ * If you want something with 'banks' to implement renaming variables apart, you probably want RobSubstitution.
  */
 class Substitution
 {
 public:
-  USE_ALLOCATOR(Substitution);
+  USE_ALLOCATOR(Substitution)
 
-  Substitution() {}
+  /**
+   * Bind `v` to `t`.
+   * If v was already present, do nothing and return false.
+   */
+  bool bind(unsigned v, TermList t) { return _map.insert(v, t); }
+  bool bind(unsigned int v, Term *t) { return bind(v, TermList(t)); }
 
-  void bind(int v,Term* t);
-  void bind(int v,TermList t);
-  void rebind(int v, Term* t);
-  void rebind(int v, TermList t);
-  bool findBinding(int var, TermList& res) const;
-  TermList apply(unsigned var) const;
-  void unbind(int var);
-  void reset();
-  bool isEmpty() const { return _map.isEmpty(); }
+  /**
+   * Bind `v` to `t`, regardless of what was there before (if anything).
+   */
+  void rebind(unsigned v, TermList t) { _map.set(v,t); }
```

Comment:
Do these two still make sense? Can't we just use `bind` without checking the result (or possibly asserting negation)?

Reply:
Ah yes, `insert` does not override.